psydac/api/ast/parser.py

```diff
@@ -508,6 +503,7 @@ def _visit_DefNode(self, expr, **kwargs):
 
         inits = []
         for k,i in self.shapes.items():
+            if not k in self.variables:continue
```

Comment:
For readability please add an empty space after the colon mark `:`

Reply:
done in [f958578](https://github.com/pyccel/psydac/pull/213/commits/f958578045dfebe7c0fd713e8e83202e2ed0b39a)